component/component.js

```diff
@@ -194,6 +197,27 @@ define('ui/components/machine/driver-%%DRIVERNAME%%/component', ['exports', 'emb
       }
    }.observes('model.%%DRIVERNAME%%Config.template'),
 
+   updateResizableOnTemplateChange: function() {
+      var templateOptions = this.get('templateOptions')
+      var selectedTemplateId = this.get("model.%%DRIVERNAME%%Config.template")
+      var selectedTemplate = templateOptions.findBy('value', selectedTemplateId)
+      selectedTemplate = selectedTemplate ? selectedTemplate : templateOptions[0]
+      this.set('templateResizable', selectedTemplate.resizable)
+      this.set('maxSizeInGb', selectedTemplate.maxSizeInGb)
+      this.set('stepSizeInGb', selectedTemplate.stepSizeInGb)
+      var templateSizeInGb = selectedTemplate.size / 1024^3
+      var minSizeInGb = selectedTemplate.stepSizeInGb
+      while(minSizeInGb < templateSizeInGb) {
+        minSizeInGb += selectedTemplate.stepSizeInGb
+      }
```

Comment:
I agree it's more dense, but it's not difficult to understand. If I wanted it to be difficult to understand I would've put `stepSize * (Math.floor(templateSizeInGb / stepSize) + !aligned)` :P